### ISSUE TITLE
Goal post-detection

### ISSUE THREAD

**Issue by manx52** [utra-robosoccer/soccerbot#191]
https://robocup.informatik.uni-hamburg.de/wp-content/uploads/2019/06/vision_paper.pdf

**Comment by Flova** [comment]
I would not recommend using the conventional goalpost detector used in this paper. We switched to a YOLO architecture last year and trained it most recently using https://github.com/bit-bots/TORSO_21_dataset to achieve a much better accuracy. It's not as fast as the model-based approach in the paper but it's worth the effort for us.
The implementation for both approaches can be found here: 
https://github.com/bit-bots/bitbots_vision
It contains the different modules for both the model-based and data-based detectors. You can also try them and change them on the fly.
YOLO implementation for multiple runtimes such as OpenVINO, OpenCV, Darkent and PyTorch: https://github.com/bit-bots/bitbots_vision/blob/master/bitbots_vision/src/bitbots_vision/vision_modules/yolo_handler.py
Model based implementaion: https://github.com/bit-bots/bitbots_vision/blob/master/bitbots_vision/src/bitbots_vision/vision_modules/obstacle.py
